### ISSUE TITLE
Rancher Desktop won't start and always chooses the Debian WSL distro for some reason

### ISSUE THREAD

**Issue by rdeusser** [rancher-sandbox/rancher-desktop#4560]
### Actual Behavior

I can't get Rancher Desktop to start. I looked in the `background.log` file and I see this:
```
2023-04-27T15:45:27.191Z: Stopping background process Win32 socket proxy.
2023-04-27T15:45:28.103Z: Error starting up: Error: C:\WINDOWS\system32\wsl.exe exited with code 1
    at ChildProcess.<anonymous> (C:\Program Files\Rancher Desktop\resources\app.asar\dist\app\background.js:1:8136)
    at ChildProcess.emit (node:events:527:28)
    at ChildProcess._handle.onexit (node:internal/child_process:291:12) {
  stdout: '',
  code: 1,
  command: [
    'C:\\WINDOWS\\system32\\wsl.exe',
    '--distribution',
    'Debian',
    '--exec',
    '/bin/wslpath',
    '-a',
    '-u',
    'C:\\Program Files\\Rancher Desktop\\resources\\resources\\linux\\bin\\docker-compose'
  ]
}
```

I use the Debian distro for something else so I can't remove it, but I'm unsure what's making Rancher Desktop choose it. Output of `wsl --list --verbose` is:
```
  NAME                    STATE           VERSION
* rancher-desktop         Stopped         2
  rancher-desktop-data    Stopped         2
  Debian                  Stopped         2
```

### Steps to Reproduce

Open Rancher Desktop via desktop icon.

### Result

No window or tray icon appears.

### Expected Behavior

Rancher Desktop should be choosing `rancher-desktop` or `rancher-desktop-data` as the distro.

### Rancher Desktop Version

1.8.1

### Rancher Desktop K8s Version

N/A

### Which container engine are you using?

moby (docker cli)

### What operating system are you using?

Windows

### Operating System / Build Version

Windows 11

### What CPU architecture are you using?

x64

### Linux only: what package format did you use to install Rancher Desktop?

None

### Windows User Only

Not sure any of this matters yet since Rancher Desktop won't even start.

**Comment by gunamata** [comment]
It looks similar to https://github.com/rancher-sandbox/rancher-desktop/issues/3785 

@jandubois , It looks we need to prioritize implementing the try/catch block as you suggested in your [comment](https://github.com/rancher-sandbox/rancher-desktop/issues/3785#issuecomment-1402913315) to gracefully ignore problematic distros.

**Comment by jandubois** [comment]
I can reproduce the problem:

* Install Debian distro (in addition to Ubuntu)

* Create an `/etc/wsl.conf` file to disable automount (probably could do this with `%USERPROFILE%\.wslconfig` as well, but didn't test this):
  ```
  root@rd-w11-test-1:~# cat /etc/wsl.conf
  [automount]
  enabled = false
  ```

* Make sure the distro is stopped, so `wsl.conf` is processed again: `wsl -t Debian`

* Verify that `automount` has been disabled:

  ```
  root@rd-w11-test-1:~# wslpath -a -u "C:\\Program Files\\Rancher  Desktop\\resources\\resources\\linux\\bin\\docker-compose"
  wslpath: C:\Program Files\Rancher Desktop\resources\resources\linux\bin\docker-compose: No such file or directory
  ```

* Verify that `%AppData%\rancher-desktop\settings.json` doesn't include `Debian` in the WSL integrations:

  ```json
  ...,
  "WSL":{"integrations":{"Ubuntu-22.04":true}},
  ...
  ```

* Try to start Rancher Desktop

No Window will be shown, and `background.log` will show the `wslpath` error from the description above.

### Required actions

* No code should be executed inside distros that have not been explicitly marked for WSL integration.

* The `getLinuxToolPath` function needs to explicitly check for errors and its callers need to deal with them. Rancher Desktop needs to display an error that WSL integration has failed for this distro (in case the user explicitly requests it to be enabled).

**Comment by mook-as** [comment]
> No code should be executed inside distros that have not been explicitly marked for WSL integration.

I disagree, sort of: we should unhook any previously done integrations in that case, but silently ignore any failures. (Think of the case where you turn off `Ubuntu-22.04` in your example.)

**Comment by jandubois** [comment]
> we should unhook any previously done integrations in that case, but silently ignore any failures

Yes, agreed. I was wondering how we could have existing integrations when the distro is not listed in `settings.json`, but I guess running factory-reset could do that.

I guess one could argue that factory-reset should undo integrations too, but that is a bigger can of worms.

**Comment by jandubois** [comment]
> I guess one could argue that factory-reset should undo integrations too, but that is a bigger can of worms.

Given that the integrations will be cleaned up on each start, deleting them as part of factory-reset is not that important.

However, they don't seem to be removed when you uninstall Rancher Desktop, which seems like a bigger problem: #5380.

**Comment by jandubois** [comment]
Please verify that any fix for this issue also covers #4966!

**Comment by naskoblg** [comment]
I had same issue with fresh install of 1.11, after I was using 0.74 long time ago on the same host. 
I was able to start rancher desktop by opening a cmd as Administrator and I have executed `rdctl factory-reset`
Hope this helps
